### PR TITLE
enable sidebar on mobile移动设备上开启侧栏

### PR DESCRIPTION
**Pull request by flashlab**
add a switch `onmobile` for `sidebar` in config.
If set, sidebar toggle will display on mobile and tablet view.

### CODE REVIEW

**Comment by ivan-nginx** [comment]
Wow, good addition. See on your blog from mobile in Mist scheme. Sidebar and b2t is work fine, but there is 1 thing: when you click on sidebar - he's slide frm right. Ok. Now, if i click on emty space (in left) - side bar is slideout. Ok. And if i click on "X" button - sidebar is slideout too. But if i want to make slideout sidebar by slide my finger from left to right, there is no effect. U understand what i meaning? Like on Android, for example. Sideout bu slide your finger from any place in sidebar radius. How i think about this option?

Also, a later will test it on Pisces scheme.

**Comment by flashlab** [comment]
@ivan-nginx It might be a a little complicate to achieve the material-like sidebar， do you have any idea to make it work?

**Comment by ivan-nginx** [comment]
@flashlab i think about it, Need some time.

**Comment by flashlab** [comment]
@ivan-nginx Any other problem? hope it'll be merged ASAP.

**Comment by ivan-nginx** [comment]
Ok, but need to strongly test it on all schemes to see on bugs.
Also, as i talk above, need to expand this feature.

**Comment by flashlab** [comment]
@ivan-nginx Works fine with Muse and Mist, while pisces has no sidebar. Should be zero-bug if the sidebar keep on the right side.

**Comment by flashlab** [comment]
#1757 implement simple swipable sidebar on touch devices.

File: layout/_macro/sidebar.swig [hunk 1]
```diff
@@ -8,6 +8,7 @@
   </div>
 
   <aside id="sidebar" class="sidebar">
+    <div id="sidebar-dimmer"></div>
```
Comment (ivan-nginx) [inline layout/_macro/sidebar.swig]: Need to add conditional import there:
{% if theme.onmobile %}

File: source/css/_common/components/sidebar/sidebar.styl [hunk 2]
```diff
@@ -36,7 +37,7 @@
 @import "site-state";
 @import "sidebar-feed-link";
 @import "sidebar-author-links";
-@import "sidebar-blogroll.styl";
+@import "sidebar-blogroll";
 @import "sidebar-nav";
 @import "sidebar-toc";
-
+@import "sidebar-dimmer";
```
Comment (ivan-nginx) [inline source/css/_common/components/sidebar/sidebar.styl]: Need to add conditional import there:
@import "sidebar-dimmer" if hexo-config('onmobile');

File: source/js/src/motion.js [hunk 3]
```diff
@@ -82,10 +82,12 @@ $(document).ready(function () {
 
   var sidebarToggleMotion = {
     toggleEl: $('.sidebar-toggle'),
+    dimmerEl: $('#sidebar-dimmer'),
```
Comment (ivan-nginx) [inline source/js/src/motion.js]: Need to add conditional import there:
```
      if (CONFIG.sidebar.onmobile) {
        ...
      }
```

File: source/js/src/motion.js [hunk 4]
```diff
@@ -82,10 +82,12 @@ $(document).ready(function () {
...
     sidebarEl: $('.sidebar'),
     isSidebarVisible: false,
     init: function () {
       this.toggleEl.on('click', this.clickHandler.bind(this));
+      this.dimmerEl.on('click', this.clickHandler.bind(this));
```
Comment (ivan-nginx) [inline source/js/src/motion.js]: Need to add conditional import there:
```
      if (CONFIG.sidebar.onmobile) {
        ...
      }
```

**Comment by ivan-nginx** [comment]
@flashlab and what about Pisces scheme? Looks like sidebar always show at the left side.

File: layout/_macro/sidebar.swig (hunk 1 above)
Comment (flashlab) [inline layout/_macro/sidebar.swig]: no need for one line I think.

File: source/css/_common/components/sidebar/sidebar.styl (hunk 2 above)
Comment (flashlab) [inline source/css/_common/components/sidebar/sidebar.styl]: not necessary consist with the context

File: source/js/src/motion.js (hunk 3 above)
Comment (flashlab) [inline source/js/src/motion.js]: no harmful I think

File: source/js/src/motion.js (hunk 4 above)
Comment (flashlab) [inline source/js/src/motion.js]: DITTO

**Comment by flashlab** [comment]
@ivan-nginx Pisces has no sidebar, so this'll not work with it.

**Comment by ivan-nginx** [comment]
@flashlab Pisces has sidebar too, but this sidebar not on same as on Mist and Muse theme.
Ok, i make additions above byself. But in next time need to do conditional import switches, because Next theme is too big with various functions and work slowly. I show u how need to do, u see it.

File: layout/_macro/sidebar.swig (hunk 1 above)
Comment (ivan-nginx) [inline layout/_macro/sidebar.swig]: https://github.com/iissnan/hexo-theme-next/commit/812c73a2fcc0be9f08dbbfd7e39e60a370772f77

File: source/css/_common/components/sidebar/sidebar.styl (hunk 2 above)
Comment (ivan-nginx) [inline source/css/_common/components/sidebar/sidebar.styl]: https://github.com/iissnan/hexo-theme-next/commit/6db9513b40d46636e8286f2854cfd3c0aaadcad6

File: source/js/src/motion.js (hunk 3 above)
Comment (ivan-nginx) [inline source/js/src/motion.js]: Ok, right.

File: source/js/src/motion.js (hunk 4 above)
Comment (ivan-nginx) [inline source/js/src/motion.js]: Ok.

**Pull request merged**